### ISSUE TITLE
KNP Breadcrumb annotation

### ISSUE THREAD

**Issue by jcornide** [mmoreram/ControllerExtraBundle#77]
Hi,
Would it make sense to add an annotation to automatically create the knp breadcrumbs? Something like:

Now:

``` php
public function listAction(
            $page, 
            $limit, 
            $orderByField, 
            $orderByDirection
    ) {

        $breadcrumbs = $this->get("white_october_breadcrumbs");
        $breadcrumbs
            ->addItem("backend.ui.home.label.title", $this->get("router")->generate("backend_home"))
            ->addItem("backend.accessory.list.label.title", $this->get("router")->generate("backend_accessories_list"));
```

With annotation:

``` php
    /**
     * Accessories list
     * @Security("has_role('ROLE_ADMIN')")
     *
     * ...
     * @Breadcrumbs({
     *   "backend.ui.home.label.title" = "backend_home",
     *   "backend.accesory.list.label.title" = "backend_accesories_list",  
     * })
     */
public function listAction(
            $page, 
            $limit, 
            $orderByField, 
            $orderByDirection
    ) {
   ....
```

I don't know if it makes sense to add this kind of annotation and "interfer" with other bundles.

**Comment by ErikTrapman** [comment]
I use KNP instead of PagerFanta (just a habit, no real reason) and i would be happy with KNP support. It would mean your listAction would look like this:

```public function indexAction(PaginationInterface $paginator)```

and the PaginatorAnnotationResolver would look sth like this:

```    public function decidePaginatorFormat(
        Paginator $paginator,
        $parameterType,
        $limitPerPage,
        $page
    ) {
        if ('Pagerfanta\Pagerfanta' === $parameterType) {
            $paginator = new Pagerfanta(new DoctrineORMAdapter($paginator->getQuery()));
            $paginator->setMaxPerPage($limitPerPage);
            $paginator->setCurrentPage($page);
        }
        if('Knp\Component\Pager\Pagination\PaginationInterface' === $parameterType){
            $knp = new \Knp\Component\Pager\Paginator();
            $paginator = $knp->paginate($paginator->getQuery(), $page, $limitPerPage);
        }
        return $paginator;
    }```

Would that be feasible? And would that help you in using KNP-breadcrumbs?

**Comment by jcornide** [comment]
@ErikTrapman to create codeblocks: https://guides.github.com/features/mastering-markdown/ look for 'php'

**Comment by ErikTrapman** [comment]
Got it. Thnx.

**Comment by ErikTrapman** [comment]
@mmoreram any thoughts on this?

**Comment by mmoreram** [comment]
@ErikTrapman Oh... yes! Why not! :)

**Comment by mmoreram** [comment]
@ErikTrapman Could you please PR it with tests? :) And ping me as soon is ready! ^^

**Comment by ErikTrapman** [comment]
lol. 
Sure, i'll see what i can do, will probably be next week though.

**Comment by JHGitty** [comment]
@ErikTrapman have you done it? :)

**Comment by ErikTrapman** [comment]
@JHGitty yes of course, it's right here: https://github.com/mmoreram/ControllerExtraBundle/pull/84
But tests fail on HHVM due to a bug that's already been fixed, but the Travis environment that does the test is still tripping over it. Maybe @mmoreram can give it another try? If it still fails i'll mark the test skipped.

**Comment by JHGitty** [comment]
@ErikTrapman The bug has been fixed for HHVM >= 3.8.0. The travis test runned with HHVM version 3.5.0. That is the reason.
